src/managers/SXCursorManager.cpp

```diff
@@ -0,0 +1,291 @@
+        return "dnd-move";
+    else if (shape == "no-drop")
+        return "dnd-none";
+    else if (shape == "not-allowed")
+        return "crossed_circle";
+    else if (shape == "grab")
+        return "hand1";
+    else if (shape == "grabbing")
+        return "hand1";
+    else if (shape == "e-resize")
+        return "right_side";
+    else if (shape == "n-resize")
+        return "top_side";
+    else if (shape == "ne-resize")
+        return "top_right_corner";
+    else if (shape == "nw-resize")
+        return "top_left_corner";
+    else if (shape == "s-resize")
+        return "bottom_side";
+    else if (shape == "se-resize")
+        return "bottom_right_corner";
+    else if (shape == "sw-resize")
+        return "bottom_left_corner";
+    else if (shape == "w-resize")
+        return "left_side";
+    else if (shape == "ew-resize")
+        return "sb_h_double_arrow";
+    else if (shape == "ns-resize")
+        return "sb_v_double_arrow";
+    else if (shape == "nesw-resize")
+        return "fd_double_arrow";
+    else if (shape == "nwse-resize")
+        return "bd_double_arrow";
+    else if (shape == "col-resize")
+        return "half-busy";
+    else if (shape == "row-resize")
+        return "sb_v_double_arrow";
+    else if (shape == "all-scroll")
+        return "fleur";
+    else if (shape == "zoom-in")
+        return "left_ptr";
+    else if (shape == "zoom-out")
+        return "left_ptr";
+
+    return std::string();
+};
+
+std::vector<SP<SXCursors>> SXCursorManager::loadStandardCursors(std::string const& name, int size) {
+    std::vector<SP<SXCursors>> newCursors;
+
+    // load the default xcursor shapes that exist in the theme
+    for (size_t i = 0; i < XCURSOR_STANDARD_NAMES.size(); ++i) {
+        std::string shape{XCURSOR_STANDARD_NAMES.at(i)};
+        auto        xImage = XcursorShapeLoadImages(i << 1 /* wtf xcursor? */, name.c_str(), size);
+
+        if (!xImage) {
+            Debug::log(LOG, "XCursor failed to find a shape with name {}, trying size 24.", shape);
+            xImage = XcursorShapeLoadImages(i << 1 /* wtf xcursor? */, name.c_str(), 24);
+
+            if (!xImage) {
+                Debug::log(LOG, "XCursor failed to find a shape with name {}, skipping", shape);
+                continue;
+            }
+        }
+
+        auto cursor = createCursor(shape, xImage);
+        newCursors.emplace_back(cursor);
+
+        if (!defaultCursor && (shape == "left_ptr" || shape == "arrow"))
+            defaultCursor = cursor;
+
+        XcursorImagesDestroy(xImage);
+    }
+
+    // broken theme.. just set it.
+    if (!newCursors.empty() && !defaultCursor) {
+        defaultCursor = newCursors.front();
+    }
+
+    return newCursors;
+}
+
+std::vector<SP<SXCursors>> SXCursorManager::loadAllFromDir(std::string const& path, int size) {
+    std::vector<SP<SXCursors>> newCursors;
+    FILE*                      f;
+    DIR*                       dir = opendir(path.c_str());
```

Comment:
god no, please, C++, https://en.cppreference.com/w/cpp/filesystem

Reply:
changed, still using fopen/fclose and FILE * since XcursorFileLoadImages expects a FILE * , one could use ifstream and deal with filedescriptors but eventually end up with fopen/fclose anyways.